File: ExpressServer/services/charactersService.js
const Character = require('../Models/characterModel');

const getAllCharacters = async () => {
    return await Character.findAll();
};

const createCharacter = async (data) => {
    const character = await Character.create(data);
    return character.id;
};

const updateCharacter = async (id, data) => {
    await Character.update(data, { where: { id } });
    return id;
};

const deleteCharacter = async (id) => {
    await Character.destroy({ where: { id } });
    return id;
};

module.exports = {
    getAllCharacters,
    createCharacter,
    updateCharacter,
    deleteCharacter,
};

File: DND.NETCore.Application/Services/CharactersService.cs
using DND.NETCore.Core.Abstractions;
using DND.NETCore.Core.Models;

namespace DND.NETCore.Application.Services
{
    public class CharactersService : ICharactersService
    {
        private readonly ICharactersRepository _charactersRepository;

        public CharactersService(ICharactersRepository charactersRepository)
        {
            _charactersRepository = charactersRepository;
        }
        public async Task<List<Character>> GetAllCharacters()
        {
            return await _charactersRepository.Get();
        }

        public async Task<Guid> CreateCharacter(Character character)
        {
            return await _charactersRepository.Create(character);
        }

        public async Task<Guid> UpdateCharacter(Guid id, string name, string description, int initiative, bool current)
        {
            return await _charactersRepository.Update(id, name, description, initiative, current);
        }

        public async Task<Guid> DeleteCharacter(Guid id)
        {
            return await _charactersRepository.Delete(id);
        }
    }
}
